### DISCUSSION TITLE
What's the correct way to manage tables and columns in production with Prisma?

### DISCUSSION THREAD

**Discussion by guicalmeida** [prisma/prisma#17575] [Q&A]
For my first backend project, I decided to create a Content Management System (CMS) with NextJS as my full-stack framework and Prisma as the ORM. As a principle, a customizable CMS must be able to programmatically create, drop and modify tables from the database; however, to my (scarce) knowledge of Prisma (and DBs in general) this can only be achieved with `prisma migrate dev` or `prisma db push`. According to the documentation, both these CLI tools should not, under any circumstances, be used in production.

I've tried to ignore the docs' warnings and run the `prisma migrate dev` programmatically with `execSync()`, but it knows when it is running under a non-interactive environment, shutting it down. Even if I was successful, it does not feel right.

This leads me to believe there's another way to manage these tables, but I can't seem to find it. The only alternative that comes to mind is to use raw SQL, which is absolutely possible but it doesn't look right, given Prisma is such a robust ORM tool.

My question then is: **how can I programmatically and safely manage relational database tables using Prisma in production?**

**Comment by guicalmeida** [comment]
[this was also asked in Stack Overflow](https://stackoverflow.com/questions/75252785/whats-the-correct-way-to-manage-tables-and-columns-in-a-postgres-db-in-producti). If you know the answer, go there as well to earn some badges and help the community!

**Reply by nurul3101** [reply]
Hey @guicalmeida 👋 

I would recommend you to have a look at this [Mental Guide](https://www.prisma.io/docs/concepts/components/prisma-migrate/mental-model) that was recently published. This describes the ideal way to use Prisma Migrate in development and production environments.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
